nest/CMakeLists.txt

```diff
@@ -31,10 +31,16 @@ if ( IS_BLUEGENE AND static-libraries )
         )
 endif ()
 
+set_target_properties( nest
+    PROPERTIES
+    LINK_FLAGS "-Wl,--no-as-needed"
```

Comment:
This flag is not supported by the Apple linker, so you need to protect this by an `if ( NOT APPLE ) ... endif()`.  I am also not sure about the linkers for more special systems, such as BlueGene and K. Maybe the best thing would be to add a test whether `--no-as-needed` is supported by the linker and only apply it if it is. Unfortunately, this does not seem straightforward in CMake, see https://public.kitware.com/pipermail/cmake/2016-January/062623.html; we need to support older CMake versions, back to 2.8.12.